runtime/domain.c

```diff
@@ -728,19 +797,19 @@ static void decrement_stw_domains_still_processing()
 }
 
 static void caml_poll_gc_work();
-static void stw_handler(caml_domain_state* domain, void* unused2, interrupt* done)
+static void stw_handler(caml_domain_state* domain)
 {
   CAML_EV_BEGIN(EV_STW_HANDLER);
-  atomic_store_rel(&done->acknowledged, 1);
   CAML_EV_BEGIN(EV_STW_API_BARRIER);
   {
     SPIN_WAIT {
       if (atomic_load_acq(&stw_request.domains_still_running) == 0)
         break;
-      caml_handle_incoming_interrupts();
 
       if (stw_request.enter_spin_callback)
         stw_request.enter_spin_callback(domain, stw_request.enter_spin_data);
+      else
+        cpu_relax();
```

Comment:
Naive question: Since `SPIN_WAIT` itself calls `cpu_relax` internally, why do we need to call it explicitly again? If the idea is to control the number of times `cpu_relax` is called, I'm wondering whether the `SPIN_WAIT` API should expose a way to do that.

Reply:
That's true. I think the logic in my head was to add some instructions given we removed `caml_handle_incoming_interrupts`. I'll remove for now.